src/pq.cpp

```diff
@@ -262,6 +262,36 @@ namespace diskann {
     }
   }
 
+  void aggregate_coords(const std::vector<unsigned>& ids, const _u8* all_coords,
+                        const _u64 ndims, _u8* out) {
+    for (_u64 i = 0; i < ids.size(); i++) {
+      memcpy(out + i * ndims, all_coords + ids[i] * ndims, ndims * sizeof(_u8));
+    }
+  }
+
+   void pq_dist_lookup(const _u8* pq_ids, const _u64 n_pts,
+                      const _u64 pq_nchunks, const float* pq_dists,
+                      std::vector<float> &dists_out) {
+    //_mm_prefetch((char*) dists_out, _MM_HINT_T0);
```

Comment:
Interesting that the original method was prefetching `dists_out`, but then was clearing the array. I agree that it seems unnecessary.